pkg/controller/machine/machine_controller_phases.go

```diff
@@ -241,6 +242,15 @@ func (r *ReconcileMachine) reconcileInfrastructure(ctx context.Context, m *v1alp
 		return &capierrors.RequeueAfterError{RequeueAfter: 30 * time.Second}
 	}
 
+	// Get and set providerID from the infrastructure provider.
+	providerID, _, err := unstructured.NestedString(infraConfig.Object, "spec", "providerID")
```

Comment:
This is now a requirement for infrastructure providers to fulfill when their status is ready

Reply:
Is this requirement documented anywhere?

Reply:
Not right now, need to tackle documentation as a different task. Have we figured out how to serve multiple versions?

Reply:
Not yet, but at the very least we should be documenting this as part of the CAEP